src/mongo.js

```diff
@@ -0,0 +1,208 @@
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+* @licend  The above is the entire license notice
+* for the JavaScript code in this file.
+*
+*/
+
+import {MongoClient, GridFSBucket} from 'mongodb';
+import {Error as ApiError, Utils} from '@natlibfi/melinda-commons';
+import {QUEUE_ITEM_STATE} from './constants';
+import {logError} from './utils.js';
+import moment from 'moment';
+
+const {createLogger} = Utils;
+
+/* QueueItem:
+{
+  "correlationId":"FOO",
+  "cataloger":"xxx0000",
+  "operation":"UPDATE",
+  "contentType":"application/json",
+  "recordLoadParams": {
+    "pActiveLibrary": "XXX00",
+    "pInputFile": "filename.seq",
+    "pRejectFile": "filename.rej",
+    "pLogFile": "filename.syslog",
+    "pOldNew": "NEW"
+  },
+  "queueItemState":"PENDING_QUEUING",
+  "creationTime":"2020-01-01T00:00:00.000Z",
+  "modificationTime":"2020-01-01T00:00:01.000Z",
+  "handledIds": []
+}
+*/
+
+export default async function (MONGO_URI) {
+  const logger = createLogger();
+  // Connect to mongo (MONGO)
+  const client = await MongoClient.connect(MONGO_URI, {useNewUrlParser: true, useUnifiedTopology: true});
+  const db = client.db('rest-api');
+  const gridFSBucket = new GridFSBucket(db, {bucketName: 'queueItems'});
+
+  return {create, query, remove, readContent, removeContent, getOne, getStream, setState, pushIds};
+
+  function create({correlationId, cataloger, operation, contentType, recordLoadParams, stream}) {
+    // Create QueueItem
+    const newQueueItem = {
+      correlationId,
+      cataloger,
+      operation,
+      contentType,
+      recordLoadParams,
+      queueItemState: QUEUE_ITEM_STATE.UPLOADING,
+      creationTime: moment().toDate(),
+      modificationTime: moment().toDate(),
+      handledIds: []
+    };
+    try {
+      db.collection('queue-items').insertOne(newQueueItem);
+      logger.log('info', 'New queue item has been made!');
+    } catch (error) {
+      logError(error);
+      throw new ApiError(500);
+    }
+
+    return new Promise((resolve, reject) => {
+      const outputStream = gridFSBucket.openUploadStream(correlationId);
+
+      stream
+        .on('error', reject)
+        .on('data', chunk => outputStream.write(chunk))
+        .on('end', () => outputStream.end(undefined, undefined, () => {
+          resolve(correlationId);
+        }));
+    });
+  }
+
+  async function query(params) {
+    const result = await db.collection('queue-items').find(params, {projection: {_id: 0}})
+      .toArray();
+    logger.log('debug', `Query result: ${result.length > 0 ? 'Found!' : 'Not found!'}`);
+    return result;
+  }
+
+  async function remove(correlationId) {
+    try {
+      await getFileMetadata({gridFSBucket, filename: correlationId});
+      throw new ApiError(400);
+    } catch (err) {
+      if (!(err instanceof Error && err.status === 404)) { // eslint-disable-line functional/no-conditional-statement
```

Comment:
Negated conditions are hard to read. Change to `err instanceof Error === false || (err.status === HttpStatus.NOT_FOUND) === false`